src/Shared/RoslynUtils/SymbolExtensions.cs

```diff
@@ -176,7 +176,18 @@ public static string GetDefaultValueString(this IParameterSymbol parameterSymbol
     {
         return !parameterSymbol.HasExplicitDefaultValue
             ? "null"
-            : SymbolDisplay.FormatLiteral((parameterSymbol.ExplicitDefaultValue ?? "null").ToString(), parameterSymbol.ExplicitDefaultValue is string);
+            : InnerGetDefaultValueString(parameterSymbol.ExplicitDefaultValue);
+    }
+
+    private static string InnerGetDefaultValueString(object? defaultValue)
+    {
+        return defaultValue switch
+        {
+            string s => SymbolDisplay.FormatLiteral(s, true),
+            bool b => SymbolDisplay.FormatLiteral(b.ToString().ToLowerInvariant(), false),
+            null => "default",
+            _ => SymbolDisplay.FormatLiteral(defaultValue.ToString(), false)
```

Comment:
I'm not familiar with `FormatLiteral()`, so sorry if this is a silly question, but if the value was a float and had a default of say, `1.0`, would this correctly emit `1.0` if the code ran on a machine where the default culture was one that uses `,` as a decimal point like French, rather than incorrectly emit `1,0`?

Reply:
What I'm thinking might be similar to this issue with the config binding source generator: https://github.com/dotnet/runtime/pull/93160

Reply:
Interesting! It looks like there is an alternate API recommended [here](https://github.com/dotnet/roslyn/issues/58705#issuecomment-1751582384) that might be helpful, although it requires pulling in different M.CA dependencies.

I'm not a huge fan of maintaining a custom implementation of `FormatLiteral` within our codebase.

I wonder if in this particular case, we can get away with not using `FormatLiteral` at all for certain types.

Reply:
Yep, I think it's worth filing an issue for this. From my experimentation, doing something like `((ParameterSyntax)IParameterSymbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax()).Default.Value.GetText().ToString()` will allow us to capture the exact constant passed as a default but we'd have to verify that nothing funky happens as a result.

Reply:
https://github.com/dotnet/aspnetcore/issues/51241